### ISSUE TITLE
help(COMPONENT): Drag and Drop

### ISSUE THREAD

**Issue by profeciak** [angular/components#19793]
#### What are you trying to do?

I have a structure like this:

<list-of-components cdkDropListGroup class = "list-of-components>
    <div *ngFor = let component of components cdkDropList (cdkDropListDropped)="drop($event)" [cdkDropListData]="component.tasks">
        <divs styling panel-group>
              <component-header>
              <component-body>
Then in my component-body I have:

<div *ngFor let task of tasks>
      <app-task>
      </app-task>
</div
and inside app-task:

<div cdkDrag cdkDrag cdkDragBoundary=".list-of-components" cdkDragLockAxis="y"
    <some dives presenting data for example task.name>
My problem is this: When Im trying to drag ma task, the dropList seems to not react. I've put ini the drop method:

        if (event.previousContainer === event.container) {
console.log("nothing changed")
        } else {
            console.log("you moved between conteners")
        }
    }}
and neither one of this console.logs is called, can someone help me with this prbolem? I just wat to move one task from it's component-body, to another component-body

#### What troubleshooting steps have you tried?

I can't see the effect of drop method, I'm thinking that my cdkDropListGroup does not react to cdkDrag inside task component

  
#### Environment

- Angular 7:
- CDK/Material:
- Chrome, Opera:
- Operating System Ubuntu:

**Comment by crisbeto** [comment]
Due to the formatting it's difficult to tell what's going on, but it looks like you have a syntax error in `<div *ngFor = let component of components cdkDropList (cdkDropListDropped)="drop($event)" [cdkDropListData]="component.tasks">`. You're missing a closing quote before `cdkDropList`.

**Comment by profeciak** [comment]
Actually it can be a mistake made during typing. I was trying to simplify the problem as far as possibe. There is not a syntax error :) here is more "solid" code
```
<div *ngFor="let component of listOfcomponents" cdkDropList (cdkDropListDropped)="drop($event)">
        <div class="panel-group" cdkDropList (cdkDropListDropped)="drop($event)" >
            <div class="panel panel-default" cdkDropList (cdkDropListDropped)="drop($event)" >
                <div class="panel-heading phase-card-header" style="display: contents">
                    <div  data-toggle="collapse">
                        <app-phase-card-header [phase]="component.data">
                        </app-phase-card-header>
                    </div>
                </div>
                <div class="panel panel-collapse collapse" cdkDropList (cdkDropListDropped)="drop($event)">
                    <div class="panel-body" cdkDropList (cdkDropListDropped)="drop($event)">
                        <app-phase-card-body [listOfTask]="component.tasks">
                        </app-phase-card-body>
                    </div>
                </div>
            </div>
        </div>
    </div>

```
I've put so much cdkDropList and drop methods because I was trying to see if any of tem would be called, unfortunately none was called (i've got a console log in drop method as stated before). I'm using the cdkDrag inside the app-phase-card-body

```
<div *ngIf="listOfTask !== null && listOfTask.length>0; else showEmpty" cdkDropList (cdkDropListDropped)="drop($event)">
    <div *ngFor="let task of listOfTask"  class="even">
        <div style="background-color: inherit " cdkDrag cdkDragBoundary=".list-of-phase" cdkDragLockAxis="y" [cdkDragData]="task">
            <app-task-row [task]="task">
            </app-task-row>
        </div>
    </div>
</div>
```

and here the drop method is called. But I don't want it here, I need it to get called in the first div of 1 part of code. As I need to get some data from task object in the 1 drop method

**Comment by crisbeto** [comment]
It's still a little hard to follow, but I think that putting `cdkDropList` everywhere might be muddying the water. The way the drag item finds which list it belongs to is by starting from itself and going up the tree until it finds a list. It's possible that the item is being assigned to a list that you aren't expecting. It may be easier to narrow it down if you provide a Stackblitz that shows the issue. You can fork this one which has everything set up already: https://stackblitz.com/angular/nqdvbrqyrde?file=src%2Fapp%2Fcdk-drag-drop-sorting-example.ts

**Comment by wagnermaciel** [comment]
Closing because this issue has been inactive for 2 weeks or more after follow up was requested.
